relay-server/src/metrics_extraction/sessions/types.rs

```diff
@@ -1,15 +1,156 @@
 use std::collections::BTreeMap;
+use std::fmt::{self, Display};
 
-use relay_common::{UnixTimestamp, Uuid};
+use relay_common::{MetricUnit, UnixTimestamp, Uuid};
 use relay_general::protocol::{
     AbnormalMechanism, SessionAttributes, SessionErrored, SessionLike, SessionStatus,
 };
-use relay_metrics::{Metric, MetricNamespace, MetricUnit, MetricValue};
+use relay_metrics::{
+    CounterType, Metric, MetricNamespace, MetricResourceIdentifier, MetricType, MetricValue,
+};
+
+use crate::metrics_extraction::IntoMetric;
+
+/// Enumerates the metrics extracted from session payloads.
+#[derive(Clone, Debug, PartialEq)]
+pub enum SessionMetric {
+    /// The number of sessions collected in the given time frame.
```

Comment:
There's no timestamp anymore.

```suggestion
    /// The number of sessions collected in a given time frame.
```